python/triton/compiler.py

```diff
@@ -1020,7 +1020,21 @@ def add_external_libs(mod, libs):
 def ttgir_to_llir(mod, extern_libs, compute_capability):
     if extern_libs:
         add_external_libs(mod, extern_libs)
-    return _triton.translate_triton_gpu_to_llvmir(mod, compute_capability)
+    ret = _triton.translate_triton_gpu_to_llvmir(mod, compute_capability)
```

Comment:
For CUDA, this check is implemented elsewhere in this file.  Should we make the change in a similar place for HIP?

Reply:
This place is one stage earlier (just after llvm ir generation) and perhaps can be promoted to upstream. Moreover, CUDA check is in online path only, but aot.py (used in offline tests) will not go into previous place.

Reply:
After thinking more about this issue, I would suggest that we mimic upstream's approach and implement this as an online-only affair.  I am not sure how this can benefit the offline stuff.

Reply:
Yes, after discussing with Alexander we got the same decision. Updated the PR.